### PR TITLE
Brightspace ESLint

### LINKED ISSUE #27: Use Brightspace eslint
Currently just extending the `eslint:recommended` settings, should be using https://github.com/Brightspace/eslint-config-brightspace

### PR DESCRIPTION
**Pull request by ryantmer**
Update to use the Brightspace ESLint rules and changes to reflect that. Fixes #27.

### CODE REVIEW

**Comment by ryantmer** [comment]
@apalaniuk you'll defs want to rebase after this is merged :P

File: d2l-course-tile.html [hunk 1]
```diff
@@ -107,14 +107,14 @@
...
-			updatePreferencesOnError: function (error) {
+			updatePreferencesOnError: function(error) {
 				error.preventDefault();
 				error.stopPropagation();
 				this._pinned = this._enrollmentEntity.getSubEntitiesByRel('preferences')[0].properties.hasOwnProperty('pinDate');
 			},
 			pinClickHandler: function(e) {
 				// Prevent the click from triggering navigation to the course
-				if(e != null) {
+				if (e !== null) {
```
Comment (apalaniuk) [inline d2l-course-tile.html]: I believe this is called without arguments when pinning from the touch menu, so 'e' will be undefined in that case? This looks like the only case where moving from equality to identity operators may cause an issue.

File: d2l-touch-menu.html [hunk 2]
```diff
@@ -2,233 +2,232 @@
...
+				this._menuItems = [].slice.call(Polymer.dom(this).querySelectorAll('d2l-touch-menu-item'));
+			},
+			attached: function() {
+				if (!this._initialized) {
+					// Only run once, as attaching to body will cause this to be called again
+					this._initialized = true;
+					this._owner = this.parentNode;
+					this._host = Polymer.dom(this).parentNode.host;
+
+					if (this.touchTarget === null) {
```
Comment (apalaniuk) [inline d2l-touch-menu.html]: This one too. For simplicity, I believe that check can be removed entirely.

File: d2l-course-tile.html (hunk 1 above)
Comment (ryantmer) [inline d2l-course-tile.html]: Ah, fair point. I'll just replace with `if (e)` in that case, which is more idiomatic to javascript anyway :)

**Comment by apalaniuk** [comment]
LGTM :+1:

**Pull request merged**